api/v1alpha3/machine_webhook.go

```diff
@@ -114,6 +116,16 @@ func (m *Machine) validate(old *Machine) error {
 		)
 	}
 
+	if m.Spec.Version != nil {
+		*m.Spec.Version = strings.TrimSpace(*m.Spec.Version)
```

Comment:
I would suggest using a new variable for this rather than modifying the object that is being validated, just in case other validations may rely on this in the future I think modifying in place could cause unexpected behaviour
```
		version := strings.TrimSpace(*m.Spec.Version)
                version = strings.TrimPrefix(version, "v")
		_, err := semver.Parse(version)
```